File: code/cryptography/src/rot13_cipher/rot13_cipher.py
# Part of Cosmos by OpenGenus Foundation
if __name__ == "__main__":
    plaintext = "This is unencrypted sample text"
    ciphertext = ""
    for ch in plaintext:
        ascii_value = ord(ch)
        if ch == " ":
            ciphertext += " "
        elif ascii_value >= 65 and ascii_value <= 90:
            if ascii_value + 13 > 90:
                ciphertext += chr(ascii_value - 13)
            else:
                ciphertext += chr(ascii_value + 13)
        elif ascii_value >= 97 and ascii_value <= 122:
            if ascii_value + 13 > 122:
                ciphertext += chr(ascii_value - 13)
            else:
                ciphertext += chr(ascii_value + 13)
    print(ciphertext)

File: code/cryptography/src/rot13_cipher/rot13_cipher.java
// Part of Cosmos by OpenGenus Foundation
public class RotN {

    public static void main(String[] args) {
        String test = "test";
		
        System.out.println(rot(test, 13)); // prints 'grfg'
        System.out.println(rot(test, 26)); // prints 'test'
    }
    
    public static String rot(String str, int n) {
        n = Math.max(0, n);
        
        String result = "";
    
        for (char c : str.toCharArray()) {
            if (!Character.isLetter(c))
                result += c;
            else {
                int b = c - 'a';
                int m = (b + n) % 26;
                result += (char) (m + 'a');
            }
        }
        
        return result;
    }
}
